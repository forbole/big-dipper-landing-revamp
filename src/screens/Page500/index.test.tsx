/* eslint-disable react/display-name */
import { act, render, screen } from '@testing-library/react';
import { motion } from 'framer-motion';
import { MockTheme } from '~tests/mocks';
import Component from '.';

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('~src/components/LayoutComponent', () => (props: object) => (
  <motion.div data-testid="Layout" {...props} />
));
jest.mock('~src/components/SectionBox', () =>
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({ main, ...props }: { [p: string]: unknown }) => (
    <motion.div data-testid="SectionBox" {...props} />
  )
);
jest.mock('~src/assets/unicorn404.svg', () => () => (
  <svg data-testid="unicorn404" />
));
// ==================================
// unit tests
// ==================================
describe('500', () => {
  it('matches snapshot', async () => {
    await act(async () => {
      global.innerWidth = 1400;
      global.dispatchEvent(new Event('resize'));
    });

    const { container } = render(
      <MockTheme>
        <Component />
      </MockTheme>
    );

    expect(screen.getByTestId('Layout')).toBeInTheDocument();
    expect(screen.getByText('500')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
