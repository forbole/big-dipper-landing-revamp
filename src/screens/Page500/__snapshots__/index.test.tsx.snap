// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`500 matches snapshot 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column nowrap;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.emotion-1 svg {
  -webkit-transform: scale(0.7);
  -moz-transform: scale(0.7);
  -ms-transform: scale(0.7);
  transform: scale(0.7);
}

.emotion-1 h3 {
  font-size: 1.2rem;
}

.emotion-1 h4 {
  font-size: 0.95rem;
}

@media (min-width:768px) {
  .emotion-1 h3 {
    font-size: 2rem;
  }

  .emotion-1 h4 {
    font-size: 1.4rem;
  }
}

.emotion-2 {
  margin: 0;
  font-size: 1.5rem;
  font-weight: 600;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  line-height: 1.167;
}

.emotion-3 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 600;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  line-height: 1.235;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    data-testid="Layout"
  >
    <div
      class="container MuiBox-root emotion-1"
    >
      <div
        class="image"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="500"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3 emotion-2"
      >
        500
      </h3>
      <p>
         
      </p>
      <h4
        class="MuiTypography-root MuiTypography-h4 emotion-3"
      >
        500Description
      </h4>
    </div>
  </div>
</div>
`;
